src/mzxml/parseBinaryDataArray.js

```diff
@@ -6,32 +8,31 @@ import { parseCvParam } from './parseCvParam';
 export function parseBinaryDataArray(node) {
   let data = node.peaks;
   let attr = node._attr;
+  let string = data._data;
+  let base64 = /^([A-Za-z0-9/+]{4})*(([A-Za-z0-9/+]{3}=)|([A-Za-z0-9/+]{2}==))?$/;
+  if (base64.test(string) === false) {
+    return {
+      data: [],
+    };
+  }
   let cvParam = parseCvParam(data._attr);
   if (!data || !attr) return [];
   let bytes = decoder(data._data, cvParam);
-
-  let kind = '';
-  if (cvParam.contentType) {
-    kind = cvParam.contentType;
-  } else if (cvParam.pairOrder) {
-    kind = cvParam.pairOrder;
-  } else {
-    throw new Error('unknown binary data type');
-  }
-
-  // console.log(buffer)
+  let kind = cvParam.contentType ? cvParam.contentType : cvParam.pairOrder;
+  let buffer = Buffer.from(bytes);
```

Comment:
Please use a typed array or a data view, not a buffer

Reply:
I guess this applies to the whole project. @josoriom could you have a look and search globally for Buffer ?